src/importlinter/application/user_options.py

```diff
@@ -20,9 +20,9 @@ def __init__(
         self.session_options = session_options
         self.contracts_options = contracts_options
 
-    def __eq__(self, other):
+    def __eq__(self, other: Any) -> bool:
```

Comment:
I would probably annotate this as `def __eq__(self, other: object) -> bool:`. https://adamj.eu/tech/2021/05/07/python-type-hints-use-object-instead-of-any/